tests/providers/opensearch/operators/test_opensearch.py

```diff
@@ -0,0 +1,126 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from opensearchpy import Document, Keyword, Text
+
+from airflow.models import DAG
+from airflow.providers.opensearch.operators.opensearch import (
+    OpenSearchAddDocumentOperator,
+    OpenSearchCreateIndexOperator,
+    OpenSearchQueryOperator,
+)
+from airflow.utils.timezone import datetime
+
+TEST_DAG_ID = "unit_tests"
+DEFAULT_DATE = datetime(2018, 1, 1)
+MOCK_SEARCH_RETURN = {"status": "test"}
+
+
+class FakeDocument(Document):
+    title = Text(fields={"raw": Keyword()})
+    author = Text()
+    published = Text()
+
+    def save(self, **kwargs):
+        return super().save(**kwargs)
+
+
+class TestOpenSearchQueryOperator:
+    def setup_method(self):
+        args = {
+            "owner": "airflow",
+            "start_date": DEFAULT_DATE,
+        }
+
+        self.dag = DAG(f"{TEST_DAG_ID}test_schedule_dag_once", default_args=args, schedule="@once")
+
+        self.open_search = OpenSearchQueryOperator(
+            task_id="test_opensearch_query_operator",
+            index_name="test_index",
+            query={
+                "size": 5,
+                "query": {"multi_match": {"query": "test", "fields": ["test_title^2", "test_type"]}},
+            },
+        )
+
+    def test_init(self):
+        assert self.open_search.task_id == "test_opensearch_query_operator"
+        assert self.open_search.opensearch_conn_id == "opensearch_default"
+        assert self.open_search.query["size"] == 5
+
+    def test_search_query(self, mock_hook):
+        result = self.open_search.execute({})
+        assert result == MOCK_SEARCH_RETURN
+
+
+class TestOpenSearchCreateIndexOperator:
+    # This test does not test execute logic because there is only a redirect to the OpenSearch
+    # client.
+    def setup_method(self):
+        args = {
+            "owner": "airflow",
+            "start_date": DEFAULT_DATE,
+        }
+
+        self.dag = DAG(f"{TEST_DAG_ID}test_schedule_dag_once", default_args=args, schedule="@once")
```

Comment:
You might consider moving the dag creation into a fixture since all these tests are repeating that block.  In addition, as a suggestion based on personal preference, I'd also inline the args.  There doesn't appear to be any reason to have them stored as a variable here.

Reply:
Good callout. Moved this to a fixture in the latest push.